### ISSUE TITLE
Include NLCD in search

### ISSUE THREAD

**Issue by Jack-Hayes** [uw-cryo/coincident#27]
Eventually want to include [NLCD](https://www.usgs.gov/centers/eros/science/national-land-cover-database) data into search rather than just ESA WorldCover for more-diverse LC types. Just raising this issue as I stumbled across the below repo for STAC searching USGS NLCD:
https://github.com/stactools-packages/usgs-nlcd

**Comment by Jack-Hayes** [comment]
Yeah the term comes from this [Malambo et al. 2021](https://www.sciencedirect.com/science/article/pii/S0034425721004314) paper on ATL08 processing evaluation but it seems "ecozone" is just a recoined term for "ecoregion" from the [RESOLVE](https://www.resolve.ngo/projects/ecoregions-world) team's 2017 global ecoregion classification https://ecoregions.appspot.com/ with 846 classes (polygons, not gridded). Available as a [GEE dataset](https://developers.google.com/earth-engine/datasets/catalog/RESOLVE_ECOREGIONS_2017) and they host a zipped shapefile (150mb) on a Google cloud.

It should be noted that this is different than the EPA's ecoregions https://www.epa.gov/eco-research/ecoregions that contains some older 2012 classifications.

I can look into the best ways to pull this data and keep you posted

**Comment by Jack-Hayes** [comment]
Hey @scottyhq I whipped up a quick notebook on ecozone access via GEE! The zones are less granular than I thought but seems like we can implement it pretty easily if needed
https://github.com/Jack-Hayes/earth-py-guide/blob/main/lulc/ecozones.ipynb

**Comment by dshean** [comment]
Hi @Jack-Hayes - I'm pretty sure you've added support for NLCD products in coincident, and you've been compiling NLCD stats for the 4-way sites. 
I didn't see anything in a quick search, however.  Did you implement a generic function for on-demand subsetting for an arbitrary polygon/bbox?  Hoping to point others to it for applications beyond coincident priorities.  Thanks!

**Comment by Jack-Hayes** [comment]
Hey David! I've been using [easysnowdata](https://github.com/egagli/easysnowdata) for NLCD products for the dataset. Eric has pretty straightforward (and great!) documentation on areal subsetting:
https://egagli.github.io/easysnowdata/examples/remote_sensing_examples/?h=nlcd#nlcd-landcover-2021

We haven't given this issue as much attention since we've been focusing so much on STAC and COGs and want to avoid the users having to have to create and set credentials for a bunch of different catalogs (e.g. GEE, AWS, EarthAccess, etc. on top of Maxar). Happy to revisit this, though
